### ISSUE TITLE
static defaultProps breaks prefer-stateless-function

### ISSUE THREAD

**Issue by evandavis** [yannickcr/eslint-plugin-react#1521]
The following is **not** caught by `prefer-stateless-function`:

```jsx
import React from 'react';
import PropTypes from 'prop_types';

export default class Thing extends React.Component {
  static defaultProps = {
     foo: ''
  };

  static propTypes = {
    foo: PropTypes.string
  };

  render() {
    return <div>{this.props.foo}</div>;
  }
}
```

However, by removing `static defaultProps`, the error is reported:
```jsx
import React from 'react';
import PropTypes from 'prop_types';

export default class Thing extends React.Component {
  static propTypes = {
    foo: PropTypes.string
  };

  render() {
    return <div>{this.props.foo}</div>;
  }
}
```

This is happening in 7.4.0.

**Comment by ljharb** [comment]
Thanks for the report!
